### PR TITLE
feat(CX-2528): My Collection artwork upload with custom artist

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-2414]

### Description

Introduces My Collection artwork upload with custom artists behind a feature flag. A custom artist name can be specified if the user skips the artist selection.

https://user-images.githubusercontent.com/4691889/166892632-e6ee9f76-1e5a-4844-aad3-deac36d2f77c.mp4

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [ ] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- My Collection artwork upload with custom artists - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-2414]: https://artsyproduct.atlassian.net/browse/CX-2414?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormMain.tsx [hunk 1]
```diff
@@ -89,11 +89,22 @@ export const MyCollectionArtworkFormMain: React.FC<
...
+                  onBlur={formik.handleBlur("artistDisplayName")}
+                  testID="ArtistDisplayNameInput"
+                  required
+                  accessibilityLabel="Artist Name"
+                  value={formikValues.artistDisplayName}
+                />
               )}
               <Input
-                title="TITLE"
+                title="Title"
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormMain.tsx:107]: This doesn't change the UI. `Input` transforms the title anyway.

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormMain.tsx [hunk 2]
```diff
@@ -246,44 +257,6 @@ const pricePaidCurrencySelectOptions: Array<{
   { label: "$ USD", value: "USD" },
   { label: "€ EUR", value: "EUR" },
   { label: "£ GBP", value: "GBP" },
-
-  // Gravity supports the following, however for the prototype
-  // we're only supporting the three above.
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormMain.tsx:251]: I don't think we need them and we can add them when we need them.

File: src/app/Scenes/Search/AutosuggestResults.tsx [hunk 3]
```diff
@@ -146,25 +147,7 @@ const AutosuggestResultsFlatList: React.FC<{
...
-                  </Text>
-                  <Text variant="md" color="black60" textAlign="center">
-                    Please try searching again with a different spelling.
-                  </Text>
-                </>
-              )
-            }
-          : null
-      }
+      ListEmptyComponent={noResults ? () => <ListEmptyComponent query={query} /> : null}
```
Comment (MounirDhahri) [inline src/app/Scenes/Search/AutosuggestResults.tsx:150]: **praise:** 👏

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/Form/artworkSchema.tsx [hunk 4]
```diff
@@ -4,9 +4,17 @@ import { trim } from "lodash"
 import * as Yup from "yup"
 
 export const artworkSchema = Yup.object().shape({
-  artistSearchResult: Yup.object()
-    .nullable()
-    .test("artistSearchResult", "Artist search result required", (value) => value !== null),
+  artistDisplayName: Yup.string().when("artistSearchResult", {
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/Form/artworkSchema.tsx:7]: **praise:** 👏

**Review by MounirDhahri — APPROVED** [review]
Great work Ole 👏

File: src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormArtist.tsx [hunk 5]
```diff
@@ -12,20 +12,41 @@ import { ArtworkFormScreen } from "../MyCollectionArtworkForm"
 export const MyCollectionArtworkFormArtist: React.FC<
   StackScreenProps<ArtworkFormScreen, "ArtworkFormArtist">
 > = ({ route, navigation }) => {
+  const enableArtworksFromNonArtsyArtists = useFeatureFlag("AREnableArtworksFromNonArtsyArtists")
   const tracking = useTracking()
+
+  const preferredCurrency = GlobalStore.useAppState((state) => state.userPrefs.currency)
+  const preferredMetric = GlobalStore.useAppState((state) => state.userPrefs.metric)
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/Screens/ArtworkForm/Screens/MyCollectionArtworkFormArtist.tsx:19]: 🙌

**Review by dariakoko — APPROVED** [review]
👏 👏 👏

**Review by MrSltun — APPROVED** [review]
Thank you for this amazing work Ole 👏 👏

**Pull request merged**